### ISSUE TITLE
🐛 Unexpected error on grit plugin with sequential patterns

### ISSUE THREAD

**Issue by golergka** [biomejs/biome#6514]
### Environment information

```block
CLI:
  Version:                      2.0.0-beta.6
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v22.14.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         bun/1.2.17

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Workspace:
  Open Documents:               0
```

### What happened?

I added this plugin to my biome configuration:

```grit
language js;


sequential {
  `$call($args)` where {
    $call <: `t.enumeration`,
    register_diagnostic(
      span = $call,
      message = "Using t.enumeration() makes server-side changes incompatible with older extension versions. Please use t.string instead."
    )
  },

  `$call<$typeArgs>($args)` where {
    $call <: `t.enumeration`,
    register_diagnostic(
      span = $call,
      message = "Using t.enumeration() makes server-side changes incompatible with older extension versions. Please use t.string instead."
    )
  }
}
```

And got these errors (a **lot** of them):

```
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue.

When opening the issue, please provide a minimal reproduction, or identify and share the file/code that triggers it. Without a way to reproduce the error, the error can't be fixed:

Source Location: /Users/runner/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/grit-pattern-matcher-0.5.1/src/pattern/state.rs:68:43
Thread Name: biome::workspace_worker_11
Message: index out of bounds: the len is 0 but the index is 0
```

### Expected result

It highlights usages of `t.enumeration()` and `t.enumeration<SomeType>()` as errors.

(BTW the same plugin but only of the first pattern works)

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
